packages/evm/contracts/adapters/pnetwork/PNetworkReporter.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.17;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IErc20Vault } from "./interfaces/IErc20Vault.sol";
+import { IPToken } from "./interfaces/IPToken.sol";
+import { PNetworkBase } from "./PNetworkBase.sol";
+
+abstract contract PNetworkReporter is PNetworkBase {
+    uint256 private constant SWAP_AMOUNT = 100;
+
+    address public immutable VAULT;
+    address public immutable TOKEN;
+
+    constructor(address vault, address token, bytes4 pNetworkAdapterNetworkId) PNetworkBase(pNetworkAdapterNetworkId) {
+        VAULT = vault;
+        TOKEN = token;
+    }
+
+    // Implement the ERC777TokensRecipient interface
+    function tokensReceived(
```

Comment:
Nope, because the IERC777Recipient interface requires an implementation for that function.
The interface is required as well because PNT is ERC777 and we need to send the reported some funds.